src/edgeDetection.cpp

```diff
@@ -37,6 +37,7 @@ void edgeDetection::enhanceEdge(frame inputFrame) {
     // Convert output cv::Mat to frame
     frame outputFrame;
     outputFrame.image = output_mat.clone();
+    outputFrame.note = "Frame processed through edge detection";  //processing note, if updated must update test
```

Comment:
Nice, but will need to rework this anyway so that notes are preserved moving through each instance. input frame should be modified and output rather than copied.
Maybe try add a float edgeLevel variable inside the frame struct definition in imageProcessor.h. Then check the edgeLevel variable rather than the string?